src/mpc-hc/SubtitleDlDlg.cpp

```diff
@@ -395,97 +209,85 @@ BOOL CSubtitleDlDlg::PreTranslateMessage(MSG* pMsg)
     }
 
-    const CAppSettings& s = AfxGetAppSettings();
-    SubtitleInput subInputToSet;
-
-    POSITION pos = m_selsubs.GetHeadPosition();
-    while (pos) {
-        const ISDb::subtitle& sub = m_selsubs.GetNext(pos);
-        CInternetSession is;
-        CStringA url = "http://" + s.strISDb + "/dl.php?";
-        CStringA ticket = UrlEncode(m_pTA->ticket);
-        CStringA args, str;
-        args.Format("id=%d&ticket=%s", sub.id, ticket);
-        url.Append(args);
-
-        if (OpenUrl(is, CString(url), str)) {
-            CAutoPtr<CRenderedTextSubtitle> pRTS(DEBUG_NEW CRenderedTextSubtitle(&pMF->m_csSubLock));
-            if (pRTS && pRTS->Open((BYTE*)(LPCSTR)str, str.GetLength(), DEFAULT_CHARSET, CString(sub.name)) && pRTS->GetStreamCount() > 0) {
-                SubtitleInput subElement(pRTS.Detach());
-                pMF->m_pSubStreams.AddTail(subElement);
-                if (!subInputToSet.pSubStream) {
-                    subInputToSet = subElement;
-                }
-            }
+    BOOL bActivate = TRUE;
+    for (int i = 0; i < m_list.GetItemCount(); ++i) {
+        if (m_list.GetCheck(i) == TRUE) {
+
+            SubtitlesInfo& subtitlesInfo = *(SubtitlesInfo*)(m_list.GetItemData(i));
+            LVITEMINDEX lvii = { i , -1 };
+            m_list.SetItemIndexState(&lvii, INDEXTOSTATEIMAGEMASK(0), LVIS_STATEIMAGEMASK);
+            HRESULT hr = S_OK;
+            subtitlesInfo.Download(bActivate);
+            bActivate = FALSE;
         }
     }
 
-    if (subInputToSet.pSubStream) {
-        pMF->SetSubtitle(subInputToSet);
-    }
-
     __super::OnOK();
 }
 
+void CSubtitleDlDlg::OnRefresh()
+{
+    m_list.DeleteAllItems();
+    m_pMainFrame.m_pSubtitlesProviders->Search(FALSE);
+}
+
+void CSubtitleDlDlg::OnAbort()
+{
+    m_pMainFrame.m_pSubtitlesProviders->Abort(SubtitlesThreadType(STT_SEARCH | STT_DOWNLOAD));
+}
+
+void CSubtitleDlDlg::OnOptions()
+{
+    m_pMainFrame.ShowOptions(CPPageSubMisc::IDD);
+}
+
 void CSubtitleDlDlg::OnUpdateOk(CCmdUI* pCmdUI)
 {
     bool fEnable = false;
     for (int i = 0; !fEnable && i < m_list.GetItemCount(); ++i) {
-        fEnable = !!m_list.GetCheck(i);
+        fEnable = (m_list.GetCheck(i) == TRUE);
     }
 
     pCmdUI->Enable(fEnable);
 }
 
-void CSubtitleDlDlg::OnFailedConnection()
+void CSubtitleDlDlg::OnUpdateRefresh(CCmdUI* pCmdUI)
 {
-    SetStatus(ResStr(IDS_SUBDL_DLG_CONNECT_ERROR));
-}
-
-void CSubtitleDlDlg::OnParse()
-{
-    SetStatus(ResStr(IDS_SUBDL_DLG_PARSING));
-    if (Parse()) {
-        LoadList();
-        CString msg;
-        msg.Format(IDS_SUBDL_DLG_SUBS_AVAIL, m_list.GetItemCount());
-        SetStatus(msg);
-    } else {
-        SetStatus(ResStr(IDS_SUBDL_DLG_NOT_FOUND));
-    }
+    bool fEnable = true;
+    pCmdUI->Enable(fEnable);
 }
 
 void CSubtitleDlDlg::OnColumnClick(NMHDR* pNMHDR, LRESULT* pResult)
 {
-    LPNMHEADER phdr = reinterpret_cast<LPNMHEADER>(pNMHDR);
+    LPNMHEADER phdr = (LPNMHEADER)(pNMHDR);

```

Comment:
(LPNMHEADER)(pNMHDR) is wrong ?

I changed it to be consistent with remaining casts in the file.